### ISSUE TITLE
Linux GOG game (Broken Sword 2) does not preserve saved data after restart

### ISSUE THREAD

**Issue by jaragunde** [Heroic-Games-Launcher/HeroicGamesLauncher#2443]
### Describe the bug

I installed Broken Sword 2, the native Linux version, from my GOG library. The game installs and runs successfully, but any changes in the game (saved games, settings) will be reset the moment you close Heroic.

I'm running Heroic 2.6.2 on a Steam Deck, installed from Flathub. I can reproduce the same problem on my Fedora laptop (same version of Heroic from Flathub).

If I run the game by double-clicking on the startup script, outside Heroic, it behaves as expected; saved games and options persist after you leave the game. If you choose to install the Windows version of the game, it also works as expected.

The problem has to do with the Flatpak sandbox. The game puts its save files into `~/.bs2`. The home dir inside the sandbox is different (by default) from the outside home: it only mirrors a few locations outside with locations inside, like the `~/Games/Heroic` dir. Anything outside these locations, like the aforementioned `~/.bs2` directory, is wiped when the Flatpak app is closed. I imagine any Linux game that puts its saves into the home directory would be affected by this.

In the case of the Windows version, the save files are put into the Wine prefix, which is also backed into the device storage, so this is not a problem for Windows games.

### Add logs

```text
(16:22:45) INFO:    [Connection]:       Connectivity: check-online
(16:22:45) INFO:    [Connection]:       Pinging external endpoints
(16:22:45) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(16:22:45) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(16:22:45) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(16:22:45) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(16:22:45) WARNING: [Backend]:          Failed to register protocol with OS.
(16:22:45) INFO:    [Connection]:       Connectivity: online
(16:22:45) INFO:    [Backend]:          User Not Found, removing it from Store
(16:22:45) INFO:    [Gog]:              Getting data about the user
(16:22:45) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --version
(16:22:45) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(16:22:45) INFO:    [Backend]:          Loading Screen Ready
(16:22:45) INFO:    [Frontend]:         Refreshing Library
(16:22:45) INFO:    [Legendary]:        Refreshing library...
(16:22:45) INFO:    [Gog]:              Getting GOG library
(16:22:45) INFO:    [Backend]:          Frontend Ready
(16:22:45) INFO:    [Backend]:          Checking for current version changelog
(16:22:45) INFO:    [Gog]:              Saved user data to config
(16:22:46) INFO:    [Gog]:              Number of library pages: 1
(16:22:46) INFO:    [Backend]:          

Heroic Version: 2.6.2 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.6

Electron Version: 22.2.0
Chrome Version: 108.0.5359.215
NodeJS Version: 16.17.1

OS: Freedesktop SDK KERNEL: 6.1.9-200.fc37.x86_64 ARCH: x64
CPU: Intel Gen Intel® Core™ i9-12900HK @3.54 GOVERNOR: powersave
RAM: Total: 62.48 GiB Available: 45.09 GiB
GRAPHICS: GPU0: Alder Lake-P Integrated Graphics Controller  VRAM: 256MB  GPU1: GA107M [GeForce RTX 3050 Ti Mobile]  VRAM: 32MB  
PROTOCOL: wayland

(16:22:46) INFO:    [Gog]:              Saved games data
(16:22:46) INFO:    [Frontend]:         No cache found, getting data from legendary...
(16:22:46) INFO:    [Legendary]:        Refreshing library...
(16:22:47) INFO:    [Gog]:              Found 0 game(s) to update
(16:22:48) INFO:    [Winetricks]:       Downloading Winetricks
(16:22:50) INFO:    [Backend]:          Starting the Download Queue
```

### Steps to reproduce

1. Install Broken Sword 2 from GOG, pick Linux version.
2. Run game.
3. Change some settings (e.g. language, screen resolution), or start a game and save it.
4. Close game.
5. Close Heroic.
6. Open Heroic and run game again.
7. Settings and savegames are gone.

### Expected behavior

Saved games and settings would be preserved.

### Heroic Version

Latest Stable

### System Information

- OS [e. g. "Ubuntu"]: SteamOS 3.4.4 and Fedora 37.

**Comment by arielj** [comment]
I'm going to close this since it's a permissions issue, we don't request the whole permissions to home because it's not needed generally, but users can change things as needed.

If there were many games needing this maybe we could add the home access permissions by default, but users might be worried about `why does heroic try to read all my files?`.

The 2 workarounds make sense and are simple to do

**Comment by jaragunde** [comment]
It makes sense, I hope that having documented the problem and the workarounds here helps people who may have this behavior in the future. Thanks!

**Comment by nnunix** [comment]
I don't think this bug should be closed at this stage. I ran into this problem yesterday with hyperlight drift from gog. I played 1 hour, and then when returning, I did no see the continue game option. I put it down to a glitch. But then it happen again. Only then I thought about searching the web and realise this was a flatpak permissions thing.

Something should be done about it because the files saved files get discarded silently and users with a less understanding on how it works, will be at a loss on what to do. By the way, for  hyperlight drift using the windows version is not possible as it crashes with all the proton versions I tried.

Maybe a warning message when users install a Linux version of game, if just setting default permissions to access home is not acceptable
